presto-spark-base/src/main/java/com/facebook/presto/spark/execution/nativeprocess/NativeExecutionProcess.java

```diff
@@ -377,4 +392,122 @@ private List<String> getLaunchCommand()
         log.info("Launching native process using command: %s %s", executablePath, String.join(" ", commandList));
         return commandList;
     }
+
+    private static class ProcessOutputPipe
+    {
+        private final long pid;
+        private final InputStreamPipe out;
+        private final InputStreamPipe err;
+
+        public ProcessOutputPipe(Process process)
+        {
+            this(process, 4096);
+        }
+
+        public ProcessOutputPipe(Process process, int bufferSize)
+        {
+            pid = getPid(process);
+            out = new InputStreamPipe(process.getInputStream(), new FileOutputStream(FileDescriptor.out), bufferSize);
+            err = new InputStreamPipe(process.getErrorStream(), new FileOutputStream(FileDescriptor.err), bufferSize);
+        }
+
+        public ProcessOutputPipe start()
+        {
+            start(out, "out");
+            start(err, "err");
+            return this;
+        }
+
+        private void start(InputStreamPipe pipe, String name)
+        {
+            Thread t = new Thread(pipe, format("NativeExecutionProcess::ProcessOutputRedirect::%s[%s]", name, pid));
+            t.setDaemon(true);
```

Comment:
Should this be a daemon thread?
I think, we want to stop the JVM from shutdown until the cpp server fully stops.

Reply:
Ah, my bad.
This is the OutputPipe not the process itself.